### ISSUE TITLE
Improve changelog to surface behavior changes

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#14147]
We've received some feedback recently about the changelogs lacking a clear indication of what behavior changes were made in a version. Particularly changes to keyboard navigation - users would like this so they can anticipate changes across minor version bumps to either mitigate or properly communicate these changes to customers of their apps.

Thinking out loud on this - to be honest, I'm not sure how we'd tackle this. The changelog is currently built by a script that looks at commit messages based on the conventional commits standard. We could extend that standard to have a new tag for behavior changes, but it would still rely on every contributor properly naming their commits (unlikely). 

I'd like to avoid manually cataloging behavior changes from version to version. It would be nice if we had something like the public-api snapshots, but for behavior. I realize this is a description of e2e tests, but how do we surface changes to e2e test flows to consumers in a way that is helpful?

----

### For anyone finding this issue that wants more out of our changelogs:

- What's missing from our changelogs from your perspective?
- Could you provide examples of things you missed or would like to see highlighted?
- Do you have ideas on how we could automate this process?

**Comment by fbarroso24** [comment]
Ideally, it'd be great if we could somehow notify or add a tag to the commits in the change log that identify any behavior changes in how users interact with an OOTB carbon component.  

For example, we recently upgraded from 1.23.0 to 1.33.2 and noticed that users can no longer tab to the `x` button to clear selections in [combobox](https://react.carbondesignsystem.com/?path=/story/components-combobox--default) etc. Users can still click on it via the mouse but to clear it via the keyboard the user can either use backspace/delete or ESC button. See https://github.com/carbon-design-system/carbon/pull/13173 for more info.

This was noticed on our end because our JEST snapshot tests changed after the upgrade and we noticed that the `tabIndex` was changed from `0` to `-1` for that `x` button.  However, for our customers who have always relied on tabbing to it via `TAB` and engaging the `Clear selection` via `ENTER` they will likely think it's a bug and submit an APAR against our product.  However, if we're able to understand these changes we can clearly communicate the behavior changes in how they interact with the OOTB carbon components.

NOTE: I know it may be hard to have it automatically add a tag to a commit.  However, in the interim even if only 25% of the folks that deliver these types of behavior changes add the tag then that'll be better than if no one adds it.

**Comment by gcattan** [comment]
May be we could have a Github action to run a `git diff` on the e2e folder when merging PRs. Then it tags the PR, which is further referenced under a new section "Behavior Changed" in your release note.

**Comment by fbarroso24** [comment]
Regarding automatically detecting behavior change in keyboard navigation, the `tabIndex` prop may be one that could be used especially if it used to be `0` and now its `-1`. In this scenario, it'd mean that you can no longer tab to it.  This would likely be the case if there is an alternate way to interact with it (e.g. using ESC button as in the [Listbox behavior change example](https://github.com/carbon-design-system/carbon/pull/13173).  Furthermore, if we're adding a `tabIndex=0`, that'd indicate that there is a new element to TAB/interact with so it'd be good to know as well since it may not be readily obvious the new capabilities in this scenario.
